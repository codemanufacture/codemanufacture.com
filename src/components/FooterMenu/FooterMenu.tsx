import * as React from 'react'
import styled from 'styled-components'
import { colors, transitions, displayDimensions, typography } from '../../theme'
import MenuItem from '../MenuItem'

const StyledMenu = styled.ul`
  display: flex;
  margin: 0;
  padding: 0;

  li {
    list-style: none;

    margin: auto;
    font-size: ${typography.paragraphSize};
    color: ${colors.fadedGrayText};

    + li {
      margin-left: 10px;

      &:before {
        content: '/';
        margin-right: 10px;
      }
    }

    a {
      color: inherit;
      font-size: inherit;
      text-decoration: none;
      transition: color ${transitions.basicTransition};

      &:hover {
        color: ${colors.white};
      }
    }

    @media (max-width: ${displayDimensions.smallPhoneSize}) {
      font-size: 0.7rem;
    }
  }

  @media (max-width: ${displayDimensions.tabletSize}) {
    justify-content: center;

    li {
      margin: initial;
    }
  }
`

const FooterMenu = () => {
  return (
    <StyledMenu>
      <MenuItem pagePath="/homepage#home" name="Home" />
      <MenuItem pagePath="/homepage#services" name="Services" />
      <MenuItem pagePath="/privacy-policy" name="Privacy Policy" />
    </StyledMenu>
  )
}

export default FooterMenu
