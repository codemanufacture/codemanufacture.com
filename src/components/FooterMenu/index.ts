import FooterMenu from './FooterMenu'

export default FooterMenu
