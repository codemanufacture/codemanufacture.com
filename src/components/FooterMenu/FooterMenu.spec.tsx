import * as React from 'react'
import renderer from 'react-test-renderer'
import FooterMenu from './FooterMenu'

describe('FooterMenu component', () =>
  it('renders correctly', () => {
    const tree = renderer.create(<FooterMenu />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
