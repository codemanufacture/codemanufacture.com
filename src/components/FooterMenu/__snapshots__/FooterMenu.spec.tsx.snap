// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterMenu component renders correctly 1`] = `
<ul
  className="sc-bdVaJa bEygpt"
>
  <li>
    <a
      href="/homepage#home"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="Home"
    >
      Home
    </a>
  </li>
  <li>
    <a
      href="/homepage#services"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="Services"
    >
      Services
    </a>
  </li>
  <li>
    <a
      href="/privacy-policy"
      onClick={[Function]}
      onMouseEnter={[Function]}
      title="Privacy Policy"
    >
      Privacy Policy
    </a>
  </li>
</ul>
`;
